mtv_dl.py

```diff
@@ -1160,6 +1180,21 @@ def download(self,
 
         return None
 
+def _guess_series_details(title: str) -> Tuple[str, str]: 
+    """Heuristics to extract season and episode information from the title."""
+    
+    # cover the "(S01/E01)" case
+    m = re.search(r"\(S(\d+)/E(\d+)\)", title)
+    if m is not None:
+        return m.group(1), m.group(2)
+    
+    # cover the "(12)" case, assume season is "01"
```

Comment:
I know there is nothing we can do about it, but I noticed shows using this annotation in cases where the season is actually different, but they don't care. 

For example `topic="Schloss Einstein" channel=ARD`:
- `Schloss Einstein (997)` is mapped to S24E23 ([Episodenguide](https://www.fernsehserien.de/schloss-einstein/folgen/24x23-folge-997-1466360))

... but there is no way to know it. Anyhow assuming 1 here may cause also invalid data in the media center.

Reply:
Good point, however I don't have a good solution for that. 

Some ideas:
- Let the user sort it out: They should only enable the `--series` flag if it works on that series.
- Hook up some internet source to get a proper mapping, this would also work for series that do not have any episode information at all.

Reply:
I would like to avoid having some dependency on a web service … those things are hard to maintain over time. 

I found another example where the used `(S01_E01)` … so it could be good to allow some non-alphanumeric chars in the middle. Even the thing with the braces may break. 

So instead of the `--series` argument, we could make the regex configurable (`--series-regex=…` or so) and enable the detection always but very strict (only the `(S01E01)` case). If `--series-regex=""` we disable the matching. If the episode number is false'ish we also do not match, but if the season number is false'ish we assume it to be "1".

What do you think?

Reply:
I haven't found a `(S01_E01)` case in the database, however the `/` in the title gets converted to a `_` in the filename.

The simplest solution first: I just replaced the `/` with an expression covering all non-alphanumeric cases to be more flexible.  (see next patch).

So the current regular expression is `"\(S(\d+)[^0-9a-zA-Z\s\)]?E(\d+)\)"`. I'm not sure if this is something we want users to configure a regex on the command line and fiddle until they got it working right, and in addition fiddling with the shell until it actually forwards all the slashes correctly. In a commercial product I would be afraid of all the user support this would create...

If we don't want users to fiddle around, they would need to patch the code (or create a bug to ask for such a change) and I would hope that over time we would cover all patterns. However that would most likely be more support work for you (or me if you forward these requests).

As this is an optional, nice-to-have feature, I guess people are not loosing anything if the series information is not extracted correctly with the currently implemented patterns.

Reply:
The regex is so complicated because it covers the `(000)` case, what we in my idea did not cover by default. 

Anyhow, i see your point that it may not be good to let users do this. For developers regexes feel very easy to use but of course most normal users are not able to write them. So better go with your initial approach to make the series parsing optional enabled with a parameter only. It will also make it easier to filter later.

Reply:
Another question … i have not seen so many examples, but can we assume this notation is always at the end of the string?

Reply:
Checked the data: no the information can be anywhere in the text (see doctests)